Package.swift

```diff
@@ -23,16 +27,21 @@ let swiftSystem: PackageDescription.Target.Dependency = .product(
     condition: .when(platforms: [.macOS, .iOS, .tvOS, .watchOS, .linux, .android])
 )
 
-let strictConcurrencySettings: [SwiftSetting] = [
-    .enableUpcomingFeature("StrictConcurrency"),
-    .enableUpcomingFeature("InferSendableFromCaptures"),
-]
+let strictConcurrencySettings: [SwiftSetting] = {
+    var initialSettings: [SwiftSetting] = []
+    initialSettings.append(contentsOf: [
+        .enableUpcomingFeature("StrictConcurrency"),
+        .enableUpcomingFeature("InferSendableFromCaptures"),
+    ])
+
+#if compiler(>=6.0)
+    if ProcessInfo.processInfo.environment["CI"] != nil {
+        initialSettings.append(.unsafeFlags(["-require-explicit-sendable", "-warnings-as-errors"]))
```

Comment:
Why are we passing `warnings-as-errors` here? We normally pass this in the CI invocations

Reply:
For at-desk development, is the answer.

Reply:
Specifically, `-require-explicit-sendable` doesn't appear to work in CI, so this dance gives us the opportunity to have the IDE itself police this if you launch it appropriately.

Reply:
Can we add this as a comment so we hopefully can remove this again in the future.

Reply:
Done.